modules/tinymce/src/core/main/ts/keyboard/MockInputEvents.ts

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright (c) Tiny Technologies, Inc. All rights reserved.
+ * Licensed under the LGPL or a commercial license.
+ * For LGPL see License.txt in the project root for license information.
+ * For commercial licenses see https://www.tiny.cloud/
+ */
+
+import Editor from '../api/Editor';
+import { clone } from '../events/EventUtils';
+
+interface SpecificsInput {
+  inputType: string;
+  cancelable?: boolean;
+  data?: null | string;
+}
+
+const mockInputEvents = (editor: Editor, specifics: SpecificsInput, event: KeyboardEvent) =>
+  () => {
+    const overrides = {
+      bubbles: true,
+      composed: true,
+      data: null,
+      isComposing: false,
+      detail: 0,
+      view: event.view,
+      target: event.target,
+      currentTarget: event.currentTarget,
+      eventPhase: event.eventPhase,
+      originalTarget: event.target,
+      explicitOriginalTarget: event.target,
+      isTrusted: event.isTrusted,
+      srcElement: event.srcElement,
+      metaKey: event.metaKey
+    };
+
+    const beforeSpecific = {
+      ...overrides,
+      cancelable: true,
+    };
+
+    const beforeInput = clone(new window.InputEvent('beforeinput'));
```

Comment:
It's because it's a dom-global and the ESlint rules prevent it from being used directly. We can add it to the allowed list which might be a better option here: https://github.com/tinymce/tinymce/blob/7fe5eb819417935aaa468848b03446200aa18a00/modules/tinymce/.eslintrc.json#L10